src/Renderer/TrailRenderer.cpp

```
@@ -20,53 +21,162 @@
+}
+
+/** ~1 canvas pixel per 500 m map scale, clamped to pre/post #2661 range. */
+[[gnu::const]]
+static double
+GetTrailThinDistance(const WindowProjection &projection,
+                     double map_scale) noexcept
+{
+  const int thin_px = std::clamp(int(map_scale / 500),
+                                 TRAIL_THIN_PIXELS_MIN,
+                                 TRAIL_THIN_PIXELS_MAX);
+  return projection.DistancePixelsToMeters(thin_px);
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)